homeassistant/components/nanoleaf/__init__.py

```diff
@@ -27,16 +29,27 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     except InvalidToken as err:
         raise ConfigEntryAuthFailed from err
 
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = NanoleafHub(hass, nanoleaf)
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {"device": nanoleaf}
```

Comment:
Consider using a dataclass instead of a `dict` as it makes it easier to type

https://github.com/home-assistant/core/blob/dev/homeassistant/components/lookin/models.py#L12